File: 프로그래머스/unrated/181929. 원소들의 곱과 합/원소들의 곱과 합.java
import java.util.stream.IntStream;

class Solution {
    public int solution(int[] num_list) {
        int multiply = IntStream.of(num_list).reduce((a, b) -> a * b).orElse(0);
		int sum = (int) Math.pow(IntStream.of(num_list).sum(), 2);

		return multiply < sum ? 1 : 0;
    }
}

File: 프로그래머스/unrated/181929. 원소들의 곱과 합/원소들의 곱과 합.kt
import kotlin.math.pow

class Solution {
    fun solution(num_list: IntArray): Int {
        val sum = num_list.sum().toDouble().pow(2).toInt()
        val multiply = num_list.reduce { a, b ->  a * b}

        return if (sum > multiply) 1 else 0
    }
}